drivers: flash: stm32 xspi and ospi: early init exit for memap mode

GeorgeCGV opened the pull request:
Exit the init earlier when OSPI or xSPI is in memory map mode. Avoid unnecessary checks and prevent pin reconfiguration
that might cause line spikes (_e.g. when pin config differs between an application and a bootloader_). Clock check beforehand is preserved (_mainly for non-mem-map mode operation as we access the register. When the clock is not enabled in mem map mode it will be noticed way earlier_).

Misc: removed `\n` from the LOG_DBG string.

GeorgeCGV committed:
drivers: flash: stm32 ospi: early init exit for memmap mode

Exit the init earlier when OSPI is in memory map mode. Avoid
unnecessary checks and prevent pin reconfiguration that might
cause line spikes. Clock check beforehand is preserved.

Remove '\n' from the LOG_DBG string.

Signed-off-by: Georgij Cernysiov <geo.cgv@gmail.com>

GeorgeCGV commented:
Compliance checks fail due to " `/bin/sh: 1: gitlint: not found`".

FRASTM reviewed (dismissed):
Probably same approach is valid for the https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/flash/flash_stm32_xspi.c

erwango commented:
> Compliance checks fail due to " `/bin/sh: 1: gitlint: not found`".

Due to recent CI change. Please rebase

GeorgeCGV committed:
drivers: flash: stm32 xspi: early init exit for memmap mode

Exit the init earlier when XSPI is in memory map mode. Avoid
unnecessary checks and prevent pin reconfiguration that might
cause line spikes. Clock check beforehand is preserved.

Remove '\n' from the LOG_DBG string.

Signed-off-by: Georgij Cernysiov <geo.cgv@gmail.com>

GeorgeCGV commented:
I rebased and added the same early exit to the xSPI driver.

erwango reviewed (approved):
Thanks for those changes

GeorgeCGV commented:
rebased once more
